WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSettingsListView.swift

```diff
@@ -0,0 +1,79 @@
+import SwiftUI
+
+/// Shows a list of domains in domain settings with a CTA to add a domain.
+struct DomainSettingsListView: View {
+    let domains: [DomainSettingsViewModel.Domain]
+    let addDomain: () -> Void
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            Text(Localization.header.uppercased())
+                .padding(Layout.headerPadding)
+                .foregroundColor(Color(.secondaryLabel))
+                .captionStyle()
+
+            ForEach(domains, id: \.name) { domain in
+                HStack {
+                    VStack(alignment: .leading) {
+                        Text(domain.name)
+                        if let renewalDate = domain.autoRenewalDate {
+                            Text(String(format: Localization.renewalDateFormat, renewalDate.toString(dateStyle: .medium, timeStyle: .none)))
+                                .foregroundColor(Color(.secondaryLabel))
+                        }
+                    }
+                    Spacer()
+                }
+                .padding(Layout.domainPadding)
+                Divider()
+                    .dividerStyle()
+                    .padding(Layout.dividerPadding)
+            }
+
+            Button(Localization.addDomain) {
+                addDomain()
+            }
+            .padding(Layout.actionButtonPadding)
+            .buttonStyle(PlusButtonStyle())
+
+            Divider()
+                .dividerStyle()
+                .padding(Layout.dividerPadding)
+        }
+    }
+}
+
+private extension DomainSettingsListView {
+    enum Localization {
+        static let header = NSLocalizedString(
+            "Your site domains",
+            comment: "Header text of the site's domain list in domain settings."
+        )
+        static let renewalDateFormat = NSLocalizedString(
+            "Renews on %1$@",
+            comment: "Renewal date of a site's domain in domain settings. " +
+            "Reads like `Renews on October 11, 2023`."
+        )
+        static let addDomain = NSLocalizedString(
+            "Add a domain",
+            comment: "Title of button to add a domain in domain settings."
+        )
+    }
+
+    enum Layout {
+        static let headerPadding: EdgeInsets = .init(top: 18, leading: 16, bottom: 6, trailing: 16)
+        static let domainPadding: EdgeInsets = .init(top: 10, leading: 16, bottom: 10, trailing: 16)
+        static let actionButtonPadding: EdgeInsets = .init(top: 18, leading: 16, bottom: 18, trailing: 23)
+        static let dividerPadding: EdgeInsets = .init(top: 0, leading: 16, bottom: 0, trailing: 0)
+        static let dividerHeight: CGFloat = 1
```

Comment:
Super nit: this doesn't seem to be used, should we remove it?

Reply:
Removed in https://github.com/woocommerce/woocommerce-ios/pull/8606/commits/82d240c3750077c457b373215009d09c66b5099f